### PR TITLE
Update cosmos sdk to last master branch

### PR DESCRIPTION
**Pull request by jgimeno**
Closes: #XXX

## Description

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/gaia/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: cli_test/cli_test.go [hunk 1]
```diff
@@ -1011,7 +996,7 @@ func TestGaiaCLIEncode(t *testing.T) {
 
 	// Check that the transaction decodes as epxceted
 	var decodedTx auth.StdTx
-	require.Nil(t, cdc.UnmarshalBinaryLengthPrefixed(decodedBytes, &decodedTx))
+	require.Nil(t, cdc.UnmarshalBinaryBare(decodedBytes, &decodedTx))
```
Comment (alexanderbez) [inline cli_test/cli_test.go:999]: @alessio this is what I meant. The internals of the test suite may break, but not the tests themselves.

**Review by alexanderbez — APPROVED** [review]
ACK. Thanks @jgimeno

File: cli_test/cli_test.go (hunk 1 above)
Comment (alessio) [reply cli_test/cli_test.go:999]: Yes, I understand your point better now, thanks!
The question is: is `decodedBytes` byte-to-byte identical to the previous output? If and only if the answer is yes, then no ABI change was introduced. Else we got an ABI breakage here

Comment (jgimeno) [reply cli_test/cli_test.go:999]: Not the same, before transaction was encoded using LengthPrefix.

Comment (alexanderbez) [reply cli_test/cli_test.go:999]: yes, and what was stated in the changelog!

Comment (alessio) [reply cli_test/cli_test.go:999]: Sure, it was mentioned in the `State Machine Breaking` stanza. My whole point is that this is worth to be mentioned in the `Client Breaking` stanza as well :)

**Comment by alessio** [comment]
However, excellent work @jgimeno!

**Pull request merged**